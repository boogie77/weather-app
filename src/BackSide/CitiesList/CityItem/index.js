import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

export default class CityItem extends Component {
  static propTypes = {
    onSelect: PropTypes.func,
    isSelected: PropTypes.bool,
    city: PropTypes.shape({
      title: PropTypes.string.isRequired
    })
  };

  onClick = () => {
    const { onSelect, city } = this.props;
    onSelect(city);
  };

  render() {
    const { city, isSelected } = this.props;
    return (
      <li
        className={classNames("list-item", { "is-selected": isSelected })}
        onClick={this.onClick}
      >
        {city.title}
      </li>
    );
  }
}
