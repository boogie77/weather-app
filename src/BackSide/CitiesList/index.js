import React from "react";
import PropTypes from "prop-types";
import { CityItem } from "./CityItem";
import "./list.css";

const CitiesList = ({ onSelect, cities, currentCity }) => {
  return (
    <ul className="list">
      {cities.map(city => (
        <CityItem
          key={city.title}
          isSelected={city.title === currentCity.title}
          onSelect={onSelect}
          city={city}
        />
      ))}
    </ul>
  );
};

CitiesList.propTypes = {
  onSelect: PropTypes.func,
  cities: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired
    })
  ),
  currentCity: PropTypes.shape({
    title: PropTypes.string.isRequired
  })
};

export default CitiesList;
