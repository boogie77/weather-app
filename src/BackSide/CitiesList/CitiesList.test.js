import React from "react";
import CitiesList from "./index";
import CityItem from "./CityItem";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

const mockCities = [{ title: "City 1" }, { title: "City 2" }];
const mockOnClick = jest.fn();

describe("<CitiesList />", () => {
  it("renders and matches the snapshot", () => {
    const wrapper = shallow(
      <CitiesList cities={mockCities} currentCity={mockCities[0]} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("calls the provided onSelect callback when clicked ", () => {
    const wrapper = shallow(
      <CitiesList
        onSelect={mockOnClick}
        cities={mockCities}
        currentCity={mockCities[0]}
      />
    );
    wrapper
      .find(CityItem)
      .last()
      .prop("onSelect")();
    expect(mockOnClick).toHaveBeenCalled();
  });
});
