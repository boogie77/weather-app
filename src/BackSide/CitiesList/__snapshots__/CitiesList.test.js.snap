// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CitiesList /> renders and matches the snapshot 1`] = `
<ul
  className="list"
>
  <CityItem
    city={
      Object {
        "title": "City 1",
      }
    }
    isSelected={true}
    key="City 1"
  />
  <CityItem
    city={
      Object {
        "title": "City 2",
      }
    }
    isSelected={false}
    key="City 2"
  />
</ul>
`;
